File: python3/1492theKthFactorOfN.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        nn = n + 1
        for i in range(1, nn):
            if n % i == 0:
                k -= 1
                if k == 0:
                    return i
        return -1

class Solution1:
    def kthFactor(self, n: int, k: int) -> int:
        for i in range(1, n + 1):
            if n % i == 0:
                k -= 1
                if k == 0:
                    return i
        return -1

File: c++/1492theKthFactorOfN.cc
#include <iostream>
#include <sstream>
#include <vector>
#include <string>
#include <queue>
#include <climits>
#include <stack>
#include <algorithm>
#include <cmath>
#include <set>
#include <unordered_map>
#include <list>
#include <unordered_set>
#include <map>
#include <functional>
#include <bitset>
#include <numeric>
#include <deque>
#include <mutex>
#include <utility>
#include <memory>
#include <cstring>

using namespace std;

static int x = [] () {ios::sync_with_stdio(false); cin.tie(0); return 0;} ();
// brute force
class Solution {
public:
  int kthFactor(int n, int k) {
    for (int i = 1; i <= n; ++i) {
      if (n % i != 0) continue;
      --k;
      if (k == 0) return i;
    }
    return -1;
  }
};

int main() {
  Solution s;
  
  return 0;
}
